scipy/fft/_basic_backend.py

```diff
@@ -142,49 +127,46 @@ def irfftn(x, s=None, axes=None, norm=None,
 
 def irfft2(x, s=None, axes=(-2, -1), norm=None,
            overwrite_x=False, workers=None, *, plan=None):
-    xp = array_namespace(x)
-    x = np.asarray(x)
-    y = _pocketfft.irfft2(x, s=s, axes=axes, norm=norm,
-                          overwrite_x=overwrite_x,
-                          workers=workers, plan=plan)
-    return xp.asarray(y)
+    return irfftn(x, s, axes, norm, overwrite_x, workers, plan=plan)
+
+
+def _swap_direction(norm):
+    if norm in (None, 'backward'):
+        norm = 'forward'
+    elif norm == 'forward':
+        norm = 'backward'
+    elif norm != 'ortho':
+        raise ValueError('Invalid norm value %s; should be "backward", '
+                         '"ortho", or "forward".' % norm)
+    return norm
 
 
 def hfftn(x, s=None, axes=None, norm=None,
           overwrite_x=False, workers=None, *, plan=None):
     xp = array_namespace(x)
-    x = np.asarray(x)
-    y = _pocketfft.hfftn(x, s=s, axes=axes, norm=norm,
-                         overwrite_x=overwrite_x,
-                         workers=workers, plan=plan)
-    return xp.asarray(y)
+    if is_complex(x, xp):
+        x = xp.conj(x)
+    return irfftn(x, s, axes, _swap_direction(norm),
```

Comment:
Currently this is being applied to all cases, I would expect this to only be used if `xp.fft` has `irfftn` but not `hfftn`. Especially for pocketfft where we have native support for `hfftn`.

Reply:
Ah yes sorry, this should be refactored to only apply when we are using the standard extension.

Reply:
Addressed https://github.com/scipy/scipy/pull/19263/commits/b75a5fb74621c241c1f643392704c038092ff745.